docs/swagger/spec.yml

```diff
@@ -184,6 +186,159 @@ paths:
+          schema:
+            $ref: '#/definitions/Layer'
+        404:
+          description: UUID parameter does not refer to a layer or the user is not able to view the layer it refers to
+          schema:
+            $ref: '#/definitions/Error'
+    put:
+      summary: Update a layer
+      tags:
+        - Layers
+      parameters:
+        - name: layer
+          in: body
+          required: true
+          schema:
+            $ref: '#/definitions/LayerSource'
+        - $ref: '#/parameters/uuid'
+      responses:
+        202:
+          description: Layer update successful (further processing needed)
+          schema:
+            $ref: '#/definitions/Layer'
+        204:
+          description: Update successful (no further processing needed)
+        404:
+          description: The UUID parameter does not refer to a layer or the user does not have access to the layer it refers to
+          schema:
+            $ref: '#/definitions/Error'
+        default:
+          description: Unexpected error
+          schema:
+            $ref: '#/definitions/Error'
+    delete:
+      summary: Delete a layer
+      tags:
+        - Layers
+      parameters:
+        - $ref: '#/parameters/uuid'
+      responses:
+        204:
+          description: Deletion successful (no content)
+        404:
+          description: The UUID parameter does not refer to a layer or the user does not have access to the layer it refers to
+          schema:
+            $ref: '#/definitions/Error'
+  /layers/{uuid}/tasks/:
+    get:
+      summary: A list of tasks associated with the layer
+      tags:
+        - Layers
+      parameters:
+        - $ref: '#/parameters/orderingBase'
+        - $ref: '#/parameters/pageSize'
+        - $ref: '#/parameters/page'
+        - $ref: '#/parameters/uuid'
+      responses:
+        200:
+          description: Paginated list of tasks associated with this layer
+          schema:
+            $ref: '#/definitions/LayerTaskPaginated'
+        404:
+          description: The UUID parameter does not refer to a layer or the user does not have access to the layer it refers to
+          schema:
+            $ref: '#/definitions/Error'
+    post:
+      summary: Start a new task associated with the layer
+      tags:
+        - Layers
+      parameters:
+        - $ref: '#/parameters/uuid'
+        - name: task
+          in: body
+          description: Task definition to start
+          schema:
+            $ref: '#/definitions/LayerTask'
+      responses:
+        202:
+          description: Task creation successful (further processing needed)
+          schema:
+            $ref: '#/definitions/LayerTask'
+        404:
+          description: The UUID parameter does not refer to a layer or the user does not have access to the layer it refers to
+          schema:
+            $ref: '#/definitions/Error'
+  /layers/{uuid}/tasks/{taskUuid}/:
```

Comment:
Yeah, I had that in my initial draft but took it out; our APIs in the past have tended to be flat because that's what DRF encourages, but a lot of those endpoints aren't very developer-friendly because if there are a lot of items we often need to construct pretty complex query strings to use the endpoint effectively. My thinking was that if we build in the most common query parameter it might improve the ergonomics a bit and we could add a `layer-tasks` endpoint later if needed. It's definitely just a shot in the dark though since we don't have any use cases yet.